src/channel_server/npc_handler.cpp

```diff
@@ -163,7 +163,27 @@ auto npc_handler::handle_npc_in(ref_ptr<player> player, packet_reader &reader) -
 }
 
 auto npc_handler::handle_npc_animation(ref_ptr<player> player, packet_reader &reader) -> void {
-	player->send(packets::npc::animate_npc(reader));
+	game_map_object npc_id = reader.get<game_map_object>();
+	size_t internal_npc_id = map::make_npc_id(npc_id);
+	
+	if (!player->get_map()->is_valid_npc_index(internal_npc_id)) {
+		// Shouldn't ever happen except in edited packets
+		return;
+	}
+	
+	uint8_t action1 = reader.get<uint8_t>();
+	uint8_t action2 = reader.get<uint8_t>();
+
+	move_path *path = nullptr;
```

Comment:
There's not really any reason for this to be a pointer.

I understand the reason why this is a pointer here and would prefer to have the `move_path` class carry some notion of whether the packet can be parsed usefully or not. That way you could have a reference to the `move_path` and pass it along to the packet function where it'll unconditionally write out the packet. The `move_path`, seeing that it can't do so, will return.

In lieu of that, I would recommend that separate codepaths based on buffer length are used if there are no changes to `move_path`.